src/lib/routing-tables.js

```diff
@@ -212,24 +213,26 @@ class RoutingTables {
 
   _findBestHopForSourceAmount (source, destination, amount) {
     debug('searching best hop from %s to %s for %s (by src amount)', source, destination, amount)
-    if (!this.sources[source]) {
+    const table = this.sources.resolve(source)
+    if (!table) {
       debug('source %s is not in known sources: %s',
         source, Object.keys(this.sources))
       return undefined
     }
     return this._rewriteLocalHop(
-      this.sources[source].findBestHopForSourceAmount(destination, amount))
+      table.findBestHopForSourceAmount(destination, amount))
   }
 
   _findBestHopForDestinationAmount (source, destination, amount) {
     debug('searching best hop from %s to %s for %s (by dst amount)', source, destination, amount)
-    if (!this.sources[source]) {
+    const table = this.sources.resolve(source)
+    if (!table) {
       debug('source %s is not in known sources: %s',
         source, Object.keys(this.sources))
```

Comment:
Does this debug statement still work? I think `Object.keys(this.sources)` needs to be replaced with `this.sources.prefixes`. Multiple instances.